src/Servers/Kestrel/test/InMemory.FunctionalTests/Http3/Http3StreamTests.cs

```diff
@@ -760,5 +761,857 @@ public async Task ResetStream_ReturnStreamError()
 
             await requestStream.WaitForStreamErrorAsync(Http3ErrorCode.RequestCancelled, CoreStrings.FormatHttp3StreamResetByApplication(Http3ErrorCode.RequestCancelled));
         }
+
+        [Fact]
+        public async Task CompleteAsync_BeforeBodyStarted_SendsHeadersWithEndStream()
```

Comment:
Are these testing the same functionality in HTTP/2 and 1.1?

Not going to thoroughly review these tests if that's the case.

Reply:
Yes